### ISSUE TITLE
[ARCHIVAL] OpenEBS

### ISSUE THREAD

**Issue by RichiH** [cncf/toc#1051]
I would like to propose an archival vote for the [OpenEBS](https://github.com/openebs) project.

As per https://github.com/cncf/toc/issues/905 , https://github.com/cncf/toc/pull/506#issuecomment-1166719790 , and https://app.slack.com/client/T08PSQ7BQ/C0MP69YF4 there have been substantial concerns about the continued health of the project for almost a year by now, with no equally substantial actions by the project to counteract or alleviate said concerns.

As per yesterday's TOC call, we are already following https://github.com/cncf/toc/pull/1050 in spirit even though it is not accepted yet. The only material difference for this process is that the two week mandatory discussion phase does not trigger immediately after informing the maintainers, the end user community must also be informed prior to the discussion phase beginning.
As this is in favor of OpenEBS, I see no issues with applying the updated process. At the active request of OpenEBS maintainers, we can also follow the old process.

CC @gila @GlennBullingham @kmova @nconnolly1 @pawanpraka1 @vishnuitta

**Comment by kmova** [comment]
Hi @RichiH. Reached out in the community slack channel. The feedback I got it is the project is actively being used for the engines that have already reached maturity - so the number of commits to those are less. With regards to Mayastor engine - there is active development and adoption.

**Comment by RealHarshThakur** [comment]
Hey @RichiH, as a past contributor of OpenEBS and current end-user at [Civo](civo.com), I'm sad to see this. 
At Civo, we rely on Mayastor to power most of our products and the performance we get from it means a lot to us. We're quite happy with Mayastor's stability and the support we get from the team too.

**Comment by vielmetti** [comment]
As I noted in the Kubernetes Slack #openebs channel, I think one of the visible concerns for the project is that there are a batch of Github issues at https://github.com/openebs/openebs/issues that have not been acknowledged in the last 45 days, and another batch (under "closed") that were closed automatically by a bot without being acked in 90 days.  Having techinical issues and bug reports and documentation errors be reported and not acked and be automatically closed is not a good look for any project.

**Comment by kmova** [comment]
Cross posting here for reference: https://github.com/cncf/toc/issues/905#issuecomment-1552336503

The active OpenEBS engine issues are being worked under: https://github.com/openebs/mayastor/issues

**Comment by kmova** [comment]
Digging a bit more with the existing maintainers - there is growing adoption of Mayastor. If the maintainers have to keep the project active - what is the information that needs to be presented?

Would preparing an annual review be the right way to approach this at this point?
https://github.com/cncf/toc/blob/main/process/sandbox-annual-review.md

**Comment by lukaszgryglicki** [comment]
Just FYI I see activity on the project.
For example:
- https://openebs.devstats.cncf.io/d/8/dashboards?orgId=1&refresh=15m.
- https://openebs.devstats.cncf.io/d/1/activity-repository-groups?orgId=1.
- https://openebs.devstats.cncf.io/d/47/github-events?orgId=1

**Comment by RichiH** [comment]
Thanks everyone! To be clear, the end result of the discussion phase does not automatically need to be a vote; that's why we have the discussion phase.

Cross-references of relevant data:
* Activity on other repos: https://github.com/cncf/toc/issues/905#issuecomment-1552336503
* End users informed yesterday: https://lists.cncf.io/g/cncf-enduser/message/514 (You need to log in, I can't see the actual email)

**Comment by MarkCupitt** [comment]
We committed to OpenEbs  not realizing that the project was at this point, very sad as it really does meet a huge need in Kubernetes. We are committed to ZFS and openebs is perfect for what we require. For us it was the zfs-local pv but we are looking at mayastor as well .. but it seems we should be looking elsewhere now .. 

Our prime concern was the lack of support and testing on recent Kubernetes releases, we are now restricted to 1.26, as zfs-localpv has a bug and it won't deploy on 1.27

May I suggest you add a note on each product page pointing out that this discussion is taking place and save some people some angst to implement a technology that may well disappear? 

No one wants to commit to core storage technology if it does not have the support and backing of the developers

**Comment by avishnu** [comment]
As one of the engineering leaders for OpenEBS, I'd like to assure that OpenEBS is very much active and the parent company is invested full-time on its growth.

1. We are making monthly releases for OpenEBS (https://github.com/openebs/openebs/releases) and Mayastor (https://github.com/openebs/mayastor/releases). We just had the latest one couple of days ago.
2. We have a very active Slack and GitHub user-community, where the team puts its best effort to resolve all user queries.
3. We hold a bi-weekly community sync-up meeting as per the CNCF standards. One can register here: https://community.cncf.io/openebs-community/

As with any long-term projects that undergo changes, we too underwent a leadership change and there could be certain areas where we were not able to focus as much as we wanted to. But we are all learning and becoming better with each passing day.

As regards to supporting the project and it's plethora of data engines, **Mayastor** is where most of the engineering is currently invested as we believe this to be a game-changer in the Kubernetes containerised storage domain. The Mayastor roadmap will be updated [here](https://github.com/openebs/openebs/blob/main/ROADMAP.md).

As regards to other data engines like Jiva, CStor and LocalPV variants, majority of them have reached feature-completeness and there are no new features being planned. We will make the best efforts to maintain these engines w.r.t. security and K8s API updates. We will always welcome contributions from our community users and open-source enthusiasts.

**Comment by RichiH** [comment]
Thanks everyone; it's clear that there's interest in the project, but also that it's not in a fully healthy state.

## Replies & questions

@RealHarshThakur can you see Civo investing engineering resources in any way?

@MarkCupitt can you, or someone else, confirm, if Kubernetes 1.27, released on April 11, 2023, is supported as of today?

@kmova I couldn't find a Slack message anywhere, could you try again please? As an alternative, we can also have a call via https://calendly.com/d/2pc-ds2-fgb/15-minute-meeting or https://calendly.com/d/z92-4t2-hqx/30-minute-meeting .

@kmova going through https://github.com/cncf/toc/blob/main/process/sandbox-annual-review.md does certainly not hurt

@avishnu OpenEBS itself seems largely stale with five releases being
* [v3.7.0](https://github.com/openebs/openebs/releases/tag/v3.7.0) May 26 2023
* [v3.6.0](https://github.com/openebs/openebs/releases/tag/v3.6.0) Mar 14 2022 - NB: same date
* [v3.5.0](https://github.com/openebs/openebs/releases/tag/v3.5.0) Mar 14 2022
* [v3.4.0](https://github.com/openebs/openebs/releases/tag/v3.4.0) Nov 16 2022
* [v3.3.0](https://github.com/openebs/openebs/releases/tag/v3.3.0) Jul 12 2022

3.7.0's only change is the adding of new maintainers with no new commits in the repository since May 26, three weeks ago as of time of writing. I went back through the commits until early 2021 and it all seemed janitorial.

https://github.com/openebs/mayastor looks a lot better, of course.

## Current state & way forward

It's good to see new maintainers being added https://github.com/openebs/openebs/commit/ac61a817c3af8e3b95edac1cccd3c0741b0806a9 

While there was some end user support, it was not a huge outpouring. This is not a deciding factor on its own, but it's still a signal. Other than the initial replies, there was no further information or specific plans for the future over the last few weeks, something I would expect to come from a project which might be archived.

There's other health signals, i.e. I would expect the license scan badge in OpenEBS' readme or the Jenkins integration for Mayastor to be working correctly, or be fixed over the last few weeks.

NB: I am using "expect" in the "that seems like a natural thing to do for a project and matches my experience in the field" not as in "[silently] demand that it must be done".

Does deprecating parts of the overall projects and focusing on other parts make sense from the project's perspective? Is there a path to getting users to contribute directly or indirectly? There's no clear opinion discernible on TOC side at the moment, so I would prefer waiting for more input and data rather than call for, or cancel, the actual vote.

**Comment by RichiH** [comment]
@kmova I know it's a top-level repo, yet it used to get a lot more attention, doc changes, etc until a time aligned with a company purchase; same as other repos.
"Change of focus/intensity by main company backing a project" is a story often told in Open Source, and certainly relevant signal; and part of why I asked if certain parts are considered feature complete or if partical deprecation and increased focus on a reduced codebase would make sense. But to also be clear: this is more signal, not a deciding factor in and as of itself.

@avishnu yes, of course! @kmova would you be willing to coordinate on OpenEBS side?

@MarkCupitt thanks! Given the spike of activity in late May 2023, maybe that issue will get fixed as well
![image](https://github.com/cncf/toc/assets/754723/0f2b8b22-f3b0-41d8-94d4-7c50b96e3c4a)

**Comment by avishnu** [comment]
>> [v3.6.0](https://github.com/openebs/openebs/releases/tag/v3.6.0) Mar 14 2022 - NB: same date

@RichiH Can I know the source of the date (Mar 14 2022) in this entry? Isn't it **Apr 27 2023** as per the release link?

**Comment by RichiH** [comment]
@avishnu I took them from https://github.com/openebs/openebs/tags -- this data is taken from the git commit's date, and the respective tags that point to said commits. I confirmed on CLI to make sure the data in GitHub is correct, and noticed that 3.5.0 & 3.6.0 point to the same commit for some reason:

````
commit abaa6904e3bea46fdad8dba2149e90f0171df219 (tag: v3.6.0, tag: v3.5.0)
Author: Niladri Halder <niladri.halder26@gmail.com>
Date:   Tue Mar 14 23:31:49 2023 +0530
````

I can't tell off-hand when the tags were created, but I can confirm that 3.4.0 points to a commit made in 2022:

````
commit c594593b6199a8d42188e5870e9090fd5dcf329f (tag: v3.4.0)
Author: Glenn Bullingham <glenn.bullingham@datacore.com>
Date:   Wed Nov 16 16:46:26 2022 +0000
````

**Comment by avishnu** [comment]
@RichiH I'd like clarify here (re-iterate what @kmova commented above) that https://github.com/openebs/openebs is NOT a code repository. It is a Landing project and a placeholder for the community guidelines. Commits happen to this repository contents only when are changes to any of these guidelines, which is not very often. As you said above, it was janitorial early 2021 as well and the company purchase event happened much later towards the end of 2021. So, there is no correlation between the two.
Please refer to https://github.com/openebs/openebs/releases for accurate information on the releases.

**Comment by RichiH** [comment]
@avishnu I know that it's a top level repository. Claiming that there's no correlation at end of 2021 is not entirely correct, though
![image](https://github.com/cncf/toc/assets/754723/3b0df606-0c7d-405e-be99-7c8cced5f5c4)

**Comment by RichiH** [comment]
TOC had an in-person offsite this week and also discussed inactive projects in general and OpenEBS in particular.

After reaching out to TAG Storage Slack channel and based on private feedback, I have asked @chira001 and Xing Yang, TAG Storage co-chairs, if the TAG could come up with specific suggestions and/or requirements of how an active and healthy OpenEBS could look like. TOC will enter a six month observation phase and make a decision based on project actions and community feedback in March 2024.

Also see https://cloud-native.slack.com/archives/C6PK4RLF7/p1694208906778189

**Comment by kmova** [comment]
Just a quick note regarding this issue. OpenEBS maintainers met with the TAG team to address the concerns. Will share an update as we solidify the action plan to address the concerns raised in this thread. At a high level, the intent is to focus on the mostly widely used engines in production and enginers that are CNCF license compliant, simplify the messaging around the active engines.

**Comment by amye** [comment]
In today's TOC meeting, the TOC requested a vote to move this to archive.

**Comment by lukaszgryglicki** [comment]
Hi, can I kindly ask to ping me when/if archival is approved? I'll update DevStats then... thanks!

**Comment by avishnu** [comment]
As OpenEBS maintainers, we'd want to assure that there is adequate activity on all the important repositories and projects. We have clarified some gaps in understanding and came up with an action plan, which has been shared with the TOC. The action plan mainly focuses on communicating clearly on the proposed status of the various OpenEBS engines. It's already work-in-progress. We are requesting for time till February to sort them. Please abstain from the voting till February.

**Comment by MarkCupitt** [comment]
Thankyou @avishnu VERY pleased to know this. 

Regards

Mark Cupitt
Global Operations Manager
Billrush

**Comment by orville-wright** [comment]
> During KubeCon Chicago, @kmova and I also found some time to sit together and we talked through the action plan. I believe Datacore / OpenEBS leadership wanted to share this plan in writing soon (relative to the KC meeting) and I understood the intention to be to have tangible goals, dates, and commitments. As mentioned on Slack with Dave Bright (@orville-wright), it would be good to have that sooner rather than later.
> 
> For transparency: TOC is internally moving to being more assertive in archival and there will be another TOC offsite middle of February.

Hi  @RichiH 
re: the OpenEBS 'Action plan' that was agreed to, Where do you want the do/info sent to ? or posted? - there are so many forums, channels etc. .that it's unclear where it needs to go to get the correct attention/coverage?

Who needs to see it? and where do they need it to be sent to / posted to? - Please advise. Myself, @avishnu @kmova will be happy to make it available.  - Thanks.

**Comment by amye** [comment]
Discussion in 1/16's TOC meeting is here: https://youtu.be/sbJVXrgRdTU?t=420

**Comment by RichiH** [comment]
Concerns about the health of OpenEBS have been raised since 2022-06-27 at the latest: https://github.com/cncf/toc/pull/506#issuecomment-1166719790

While there have been multiple and repeated statements of intent to revive the project and/or to archive stale portions of it, we saw little action over the last ~1.5 years.

Please note that if the archival vote is successful, it does not prevent anyone from working on OpenEBS and re-applying for sandbox level; matter of fact, that would be the best possible outcome of an archival.

**Comment by orville-wright** [comment]
Thanks @RichiH for you feedback,

re: the stale portions of the project (Jiva, cStor, NFS-Provisioner).
The project team doesn't have any objections to this concept and we are in support of it as it makes good sense.
This is definitely where the KubeCon Chicago discussion (with Alex and Xing) were mostly centered on. (note: I was in that meeting). 

- The active part of the project (Mayastor) is very  active, and is where most of the user interest, coding & community activity are focused and is happening... becasue we see that users want NVMe/SSD/.Flash capabilities. The OpenEBS Mayastor project component is the part of the project that is highly specialized and optimized in that technology, those capabilities and that feature set.... whereas these features don't exist in the "stale portions".

A key dilemma for the team is that we don't fully understand how (physically) CNCF would split-out and archive off the stale portions while supporting & enabling the modern/active sub-project "Mayastor", enabling it to live-on and continue on as primary project of OpenEBS ? 
- Can  you help us understand how the logistics of this physically would be done ?? and what it looks like?? and how such project surgery happens ? 

After meeting with Alex and Xing in Chicago, we've had project team, numerous meetings and many folks felt that 'physically slicing off the stale projects' ... was not what CNCF was asking for; but now... you guys have clarified that this is what you want. 
- We support that, but wouldn't want the growing OpenEBS Mayastor community to be negatively affected as this is were all the interest, excitement, energy and activity is for OpenEBS.  
- So we're a bit confused about the logistics and execution on the plan.

~Dave

**Comment by Zman2356** [comment]
To the ToC and the OpenEBS user community, Please accept our sincere apologies for putting the OpenEBS project in a position of an archive vote.  The product commitment has been there all along, but we lacked the proper community engagement which will be corrected immediately.  Please view this short explainer video https://share.vidyard.com/watch/Nw7XBqbCDgraxgiVrbx8pX? 
Dave Zabrowski, CEO of DataCore Software

**Comment by edrob999** [comment]
**I was the person at KubeCon in Chicago who promised the project team would address issues.**
Here is what's going on. We committed to four things:
- "Cleaning up" the older storage engines, adding readme/roadmap, making sure the issues are all deal with (project team had concentrated on Slack, more than GitHub), updating all the documentation
- We committed to meaningful progress by next KubeCon (March)
- As we saw this thread, the team accelerated things. Last couple days we put the readme/roadmap on GitHub and made sure the issues are tagged properly. The cleanup of the older engines is going to take a few more weeks - but we'll try to do this as quickly as possible. Docs are super important - and it will take a little more time. 

OpenEBS is a very active project. I now work for DataCore, and work alongside 20 engineers fulltime on the OpenEBS project, and they are very passionate and proud to work on such a widely used project. Looking at the developer activity counts on cncf.io, outside DataCore another 332 people have contributed to the project in the last year. We know a lot of companies, maybe thousands, use the software in production, including Microsoft  - Microsoft Azure Container Service is derived from OpenEBS. This project is alive and well.

**Communication**
Its clear we have a communication problem. RichiH - I've tried connecting with you on LinkedIn, I would send you my email/phone, so you can reach if you're not getting through. There are so many slack channels! For me - I'm new here, I'm simply not seeing the questions you're posting, but it doesn't mean I'm not listening.

If the ToC decides to keep the project from being archived, I hope we can designate a single channel - whether its a specific Slack Channel, or GitHub or something else that we set as the official channel, so this doesn't happen again.


I also hope there is a way we can keep the OpenEBS project from being archived. The project team feel awful, they put their passion + energy into the project, and want to see it move towards incubation and graduation. They are upset that OpenEBS could be archived instead.

**Comment by kmova** [comment]
Just my 2c. 

I can see that the code is actively maintained. 
https://openebs.devstats.cncf.io/d/2/commits-repository-groups?orgId=1&var-period=w&var-repogroups=All&from=now-4y%2Fy&to=now

And anecdotally speaking to different folks - there were atleast two large production users. I wonder if CNCF could actually make sometime to talk directly with those users around the responsiveness of the project maintainers before taking the vote. 
* 
It is really unfortunate to see a project that is actively developed and used in production go through this vote due to the lack of community engagement. 

(The optimist in me would like to give this https://github.com/cncf/toc/issues/1051#issuecomment-1902074535 a chance. What is another 3 months - to check if the team will come through, instead of rushing the vote asap.)

**Comment by lukaszgryglicki** [comment]
@kmova you mentioned DevStats - actually runs using OpenEBS as a storage option.

**Comment by RichiH** [comment]
It is clear that there are people who care deeply about OpenEBS and it's also clear that there are large/important users and use cases and no clear replacement. Yet, it's also evident that there's a repeating pattern of a flurry of communication and commitments every time TOC moves towards implementing the policies in place, with a seemingly inevitable fall-off shortly after. We have gone through this cycle repeatedly over the last 19 months.

While there are several possible "official" ways to communicate specific and committed plans -- such as Slack, this GH issue, https://github.com/cncf/toc/issues/905, or TAG Storage, none have been used to communicate said plans. As communicated previously, GitHub is preferred as it's public, searchable, and time-stamped.

The outcome of the TOC vote is not certain either way. No matter the outcome of the vote, a good outcome for the project and the wider community would be a communication of committed plans and then their subsequent execution.

Again, if the archival vote succeeds, the door does not close. While trademark and marketing are impacted, the GitHub repositories are not. Development can, and arguably should, continue. This can also be seen as a forcing function for both the management of existing contributors and large users who are currently not contributing; as a loud cry for help and investment.

An archived project can re-apply for sandbox at any time. As communicated previously this is the best possible outcome of a successful archival vote.

**Comment by Zman2356** [comment]
@RichiH, you are absolutely correct in your observation as we simply have not communicated as we should have.  That will not happen again as I am now holding weekly all hands stand ups and funding a full time community lead (in addition to 20 dedicated devs). I will ask the dev team to post the detailed plan to GitHub as you request.  I hope the ToC will see that while the communication has not been consistent, the commitment to the project has been.

**Comment by avishnu** [comment]
[OpenEBS - Project Health Improvements.pdf](https://github.com/cncf/toc/files/14050839/OpenEBS.-.Project.Health.Improvements.pdf)
Sharing the plan and the progress towards the same.
Thank you.

**Comment by RichiH** [comment]
Thanks. For convenience, https://docs.google.com/document/d/1QyWBa1ksJTyv6RawPrD9uJDqz4DEMIhSuTU5xs6OjFI/edit is the direct link the the GDoc underlying the PDF.

**Comment by edrob999** [comment]
@RichiH Agree that OpenEBS is a useful project, used by a lot of Kubernetes users in production. Also agree the communication needs improvement, and requests haven't been dealt with. As I've watched the project and talked to some of our members, people in the OpenEBS community contribute because they're interested in storage + Kubernetes. Many are not confident to communicate with CNCF on project issues.

I (Ed Robinson) am happy to steward improving this, to understanding the CNCF guidelines, making sure the project agrees a path compliant,  communicating with CNCF + resolving issues. I'll also set up a process so that the OpenEBS community can vote for a new steward.

Moving the project to archive doesn't solve the communication problem.
It just means someone else has the to deal with the problem later. "Kicking the can down the road"

I'm hoping we can keep the project from being archived, and I will actively act as steward to make sure we are communicating, responding, and dealing with requests.

Ed

**Comment by TheFoxAtWork** [comment]
👋🏻 Hello everyone, in order to avoid future confusion as to the outstanding requests for OpenEBS and the impetus behind moving to an archive vote, the TOC has pulled together the following information to share with the project and all interested parties.

## Decision to move to Archive Vote

The TOC has made the decision to bring the OpenEBS project to a vote for archival in keeping with our [Archive](https://github.com/cncf/toc/blob/main/process/archiving.md) process; [end users were notified on May 17th, 2023](https://github.com/cncf/toc/issues/1051#issuecomment-1553504989) of the proposal to archive. 

### Archive process

The Archive process encompasses the TOC evaluating different indicators that were observed by community members and brought to the TOC's attention. Project health is one but not the only indicator the TOC weighed in moving this project to a vote for archive. While the TOC does anticipate project’s to shift their scope in response to contributors’ and adopter feedback, we expect Sandbox projects to experiment and make progress towards incubation, documenting their decisions transparently, demonstrating alignment with [CNCF Values](https://github.com/cncf/foundation/blob/main/charter.md#3-values), and completing or responding to recommendations provided by the TOC and TAGs to enable the project to be successful within the CNCF.

### Past engagement with OpenEBS

The TOC has engaged the OpenEBS project on several occasions since this issue and [its predecessor](https://github.com/cncf/toc/issues/905) were opened on the repo. Issue #905 was filed after [a low point of commit activity during the acquisition of Mayadata](https://openebs.devstats.cncf.io/d/2/commits-repository-groups?orgId=1&var-period=w&var-repogroups=All&from=now-4y%2Fy&to=now). It is anticipated companies who donate projects to CNCF could become acquired, which may result in decreased or stalled activity, occasionally withdrawing from a project altogether. 

Since OpenEBS was accepted into the CNCF as a Sandbox project, numerous issues with the project have occurred, in varying states of resolution with limited tracking. The OpenEBS project also made several, significant changes to their focus, most notably concerning the storage engines, and finally with the control plane, eventually concentrating its development efforts around Mayastor.

Over several years of discussions with the project, its maintainers, and TAG Storage, the TOC has evaluated the project’s governance, documentation, activity, and general practices against those we expect of a Sandbox project. The [last Annual review the project provided to the TOC](https://github.com/cncf/toc/pull/685/files), prior to removing manual annual reviews (2023) in favor of automation (coming 2024), was in 2021. The [project applied to incubation level](https://github.com/cncf/toc/pull/506) in 2020, received feedback from TAG Storage in August 2020, January 2021, and March 2022, but ultimately paused the incubation application in June 2022 and subsequently closed in May 2023 after highlighting continued areas of improvement needed by the project to be ready for incubation.

### Outstanding problems

In addition to prior requests for resolution of feedback and action items, the TOC has reviewed the project’s governance practices and documentation, determining that:

- the documentation provided to contributors and adopters is out of date, in many cases years out of date

- there are inconsistencies and conflicts in statements regarding the current state and stability of its 4 storage engines

- non-conformance to the project’s own defined governance regarding adding and removing maintainers

#### Sandbox expectations

Further, in re-evaluating OpenEBS against expectations of projects in a sandbox status (refer to the [Sandbox Annual Review guide introduced with the TAGs in 2023 before removing manual annual reviews](https://docs.google.com/document/d/1cE9zutKh3ZGzEzZr5CZ_ck66JJ1J5mLeNYELVLbxf-Q/edit)) the following is observed:

- The project scope and goals remain consistent at a high level, with deviation from when accepted into sandbox as a result of experimentation
- Community development and growth has stalled, the last meeting notes occurred in [August of 2023](https://hackmd.io/tRmnyg9AQPShxFAXOyFpSA?edit) citing cancellation due to low attendance, prior community meetings focused heavily on Mayastor almost exclusively
- Appropriate project governance is not being consistently adhered to
- Long term planning exists, [notably focused on Mayastor](https://github.com/openebs/openebs?tab=readme-ov-file#roadmap-as-of-jan-2024)
- Development is ongoing and progressive towards versioned releases however much of the supporting documentation for OpenEBS has not been updated to coincide with OpenEBS releases. For example, [OpenEBS of K8S](https://github.com/openebs/openebs/blob/main/k8s/README.md) has not been updated in 5 years over the course of roughly 25 releases
- Adoption is occurring: in the past two years 6 adopters were added, (less than in previous years) but it is happening
- Self-awareness for pursuing incubation has not resumed within the project since the application was closed in May 2023. Having previously applied and feedback provided by the TOC sponsor, it is reasonable to expect the project to track those actions, make and demonstrate progress towards their completion, and reapply at a later date.

### What to expect next

Typically, the findings and recommendations from sandbox annual reviews are filed with the project as an issue on their repo for tracking to completion, however given the continued recommendations over the past 3 years by TAG Storage and the TOC to bring the project back into alignment with these expectations, the TOC has called a vote to archive OpenEBS.

This vote has been called to place the project in a state of archive. For more information on the expectations of an archive state in CNCF please refer to our [Archive Status section](https://github.com/cncf/toc/blob/main/process/archiving.md#archived-status), please note that archived projects may continue development in its repositories, ideally with a goal to reapply.

#### If the Archive vote passes

In the event the Archive vote passes, the following recommendations are made to OpenEBS and Mayastor, should they choose to reapply:

- OpenEBS may re-apply to sandbox at a later date provided the following actions are completed:
  - Demonstrated alignment with the Sandbox expectations listed above
  - Roadmap to expresses planned future work of the entire OpenEBS project
  - Demonstrated alignment and exercise of the project’s own governance, particularly for contributions, and especially for maintainer selection, turnover, and maintainer onboarding.
    - It is expected that regardless of employer, individuals proposed for maintainer roles on a project follow the same process.
- Mayastor may formally re-apply as a sandbox project, if interested, contingent upon the above actions also being completed unique to the Mayastor project.

#### If the Archive vote does not pass

In the event the Archive vote does not pass, we fully expect the OpenEBS project to file an issue or series of issues on their repo, cross linked/referenced with this Issue so the TOC may monitor and track the above recommendations and the expectations to conclusion with the OpenEBS project, with substantive progress by the end of August 2024.

## Final Note

We appreciate the OpenEBS project and its maintainers engaging on this issue recently, and thank everyone for their insights, observations, and efforts in bringing this discussion to closure.
	
As a reminder, the TOC maintains several official communication mediums by which project maintainers, contributors, and interested parties may reach out to the TOC. We may be reached on slack in the #toc channel, by email at cncf-toc@lists.cncf.io or cncf-private-toc@lists.cncf.io (for sensitive matters), through our Repo by filing an issue, or by joining our twice monthly public meetings.  This ensures all TOC members are kept apprised of discussions, statuses, and commitments of projects and the TOC. We have several CNCF staff that support the TOC in monitoring these communication methods. Directly contacting TOC members individually is generally discouraged as we perform this function in addition to our regular work and such contact may be lost amid day-to-day tasks, illnesses, holidays, or other events.

**Comment by edrob999** [comment]
Thanks @TheFoxAtWork for this info. **It is the project team's request that OpenEBS is NOT archived.** We commit to resolving the issues you have identified, using the cross linked/referenced process you suggest, and making substantive progress before end of August 2024.

**Comment by edrob999** [comment]
After the OpenEBS project status was changed to "Archive", we discussed what happened and decided as a team we will adopt the changes recommended by TOC members, and follow the advice to formally re-apply as a sandbox project once we are ready. We have about OpenEBS 700,000 users that rely on the project, we owe it to them to move as quickly as we can. We're tracking the changes in a GitHub parent issue @avishnu cross-linked above.

**Comment by RichiH** [comment]
@edrob999 thanks for the update. This seems to be great plan to go forward with. Looking forward to OpenEBS resubmitting in the future!

**Comment by Ornias1993** [comment]
@RichiH and @TheFoxAtWork I want to highlight that while OpenEBS is trying to get readmitted, this readmittence should not be accepted within any reasonable timeframe.

During their changes to become readmitted, they managed to break their Helm Charts, affecting MANY production users, multiple times by making incredible amateuristic mistakes. They managed to do so 3 times in *just* 2024.

There also seems to be no reflection what so ever on how bad these mistakes where and the ability to take responsibility for these significant mistakes, seems to be completely lacking.

In short:
OpenEBS has shown they are not ready to produce production ready software packages.
Which should invalidate them for readmittance for a significant time and stay archived.

**Comment by edrob999** [comment]
Hello again @Ornias1993. In response to your other suggestions: No we are not going to fire the person responsible. No this was not done by a beginner intern. If you want to talk, just ping me. We're all on the same side, and sometimes talking helps us understand each other better.

@Ornias1993 is correct. What happened is: On the TOC's recommendation, the OpenEBS maintainers [gave notice](https://github.com/openebs/openebs/issues/3709) we would be archiving engines/repos that were deprecated/no longer maintained. We then archived about 66% of the project, and sought to keep the archived versions installable + working. This stuff is complicated! When we archived the engines some helm charts had problems. [You can see](https://github.com/openebs/zfs-localpv/issues/517) the team jumped on it quickly, and made fixes. 

We didn't want this to happen. We responded really quickly when it did. Every member of our community is important - I'm sorry for your hurt @Ornias1993, let me know if you want to talk thru it.  
Ed

**Comment by Ornias1993** [comment]
> No this was not done by a beginner intern

I was indeed made aware of it.
When its done by someone that is not an intern, making beginner mistakes and none of the maintainers catching them, that is even more worrysome and hence my statement that the project should not be readmitted anytime soon.

> the OpenEBS maintainers https://github.com/openebs/openebs/issues/3709 we would be archiving engines/repos that were deprecated/no longer maintained

You indeed posted such a notice and keep pointing to that.
however:
- ZFS Local-PV is compeletely not on that list of being an "archived engine", so the argument is completely invalid

>  We then archived about 66% of the project, and sought to keep the archived versions installable + working.

ZFS Local-PV wasn't an achieved version on that list

> This stuff is complicated!

Actually, realising that nuking a helm-repo index file causes breakage for users is absolutely minimally trivial.

> the team jumped on it quickly, and made fixes.

I appreciate the guy breaking your helm charts jumping on it within a few days to fix his mistakes again.
That doesn't make it more acceptable.

> We didn't want this to happen

I sincerely hope so.
But you've staff members making beginner mistakes, other maintainers not carefully reviewing them (or not being able to, skill wise), not testing things locally in any way and then there is *always* an excuse coming up.

At the very team the staff member involved should've been put on a heavy coaching and training course.

But the fact there is always an excuse for these entry-level mistakes, show that OpenEBS and Mayadata is *still* not taking actual accountability. This time it's, yet again, pointing at an unrelated announcement about things being archived.

Which:
A. Had nothing to do with the first 2 times your staff broke the production helm charts
B. Had nothing to do with non-archived helm charts

---

I'm not hurt or angry in any way, I just objectively came to the conclussion that OpenEBS is not ready for readmittance due to the way it is actually handling its attempt at being readmitted.

**Comment by avishnu** [comment]
@Ornias1993 Don't you feel you are being unreasonably rude here? Appreciate if you show bit more gratitude for open-source developers.

We acknowledged the mistakes and tried to contain the impact within reasonable time-frames by revoking the affected changes and providing the fix there-after. The users who brought this to our attention, were satisfied with our responses.

Open-source is about collaboration. If you could have helped in the review of the change request which led to the ZFS LocalPV Helm chart bug, we might have avoided the situation. We always appreciate and value constructive insights from external contributors, including criticism. Sorry to say, your words are hurtful.

You willing, we'd be happy to have you help us on Helm-related changes in the future.

**Comment by Ornias1993** [comment]
> @Ornias1993 Don't you feel you are being unreasonably rude here? Appreciate if you show bit more gratitude for open-source developers.

No I do not feel i'm unreasonable.
Mayadata, its employee(s) and the OpenEBS volunteers have shown 3 times in a few months, to not being able to maintainer production-grade helm charts. I'm not going to sugar coat that fact.

A post about OpenEBS being not being ready for readmittence is not the place "to show gratitude", i'm not going to sugarcoat that with "american service" level nonesense by first writhing a whole piece about how gratefull I am.

> We acknowledged the mistakes

"ohh yes this is a mistake we need to fix"

is not the same as

"Our procedures where sub par and our review was completely lacking. We've done X, Y and Z to prevent this in the future"

The later is *actually* taking accountability.

Just admitting something was a mistake, then making 2 similair entry-level mistakes in the months following, does not show actual understanding of the severity.

> The users who brought this to our attention

I was not, nor where our 20-35k users.
This is *again* skirting around the severity of the issue by all sorts of arguments that are inherently flawed.

> if you could have helped in the review of the change request which led to the ZFS LocalPV Helm chart bug, we might have avoided the situation

It's not like I get daily updates on each PR opened on each software package I use...
But even so, I maintain close to 800(!) helm charts. Time is limited.

As a downstream user, I trusted OpenEBS as a trustworthy upstream.
That trust was broken 3 times in the span of a few months.

Without any *actual* accountability taken, which means: Actually reflecting on mistakes and how they are going to be prevented in the future.

> Sorry to say, your words are hurtful.

Your, as in OpenEBS maintainers, actions where factually damaging.
I find that more important.

---

Simply put:
What OpenEBS is doing here, even while I highlighted *multiple times* how they can show accountability, is again-again-and-again playing all sorts of "social cards" about how hurt they are and how they "tried", how "they communicated so well", how "other users are fine" etcetc.

Instead of:
"We ducked this up badly, we've done x, y and z to prevent this in the future".
Which should've done the first time a entry-level helm mistake hit the production builds.

Even after these 3(!) big mistakes in a few months they could've shown said accountability.
But they won't.

It isn't about feelings, mine nor theirs, it's about trustworthy software.
